client/src/main/scala/io/delta/sharing/client/DeltaSharingFileSystem.scala

```diff
@@ -22,37 +22,78 @@ import java.util.concurrent.TimeUnit
+import org.apache.http.impl.conn.{DefaultRoutePlanner, DefaultSchemePortResolver}
+import org.apache.http.protocol.HttpContext
 import org.apache.spark.SparkEnv
 import org.apache.spark.delta.sharing.{PreSignedUrlCache, PreSignedUrlFetcher}
-import org.apache.spark.network.util.JavaUtils
 
 import io.delta.sharing.client.model.FileAction
-import io.delta.sharing.client.util.{ConfUtils, RetryUtils}
+import io.delta.sharing.client.util.ConfUtils
 
 /** Read-only file system for delta paths. */
 private[sharing] class DeltaSharingFileSystem extends FileSystem {
+
   import DeltaSharingFileSystem._
 
   lazy private val numRetries = ConfUtils.numRetries(getConf)
   lazy private val maxRetryDurationMillis = ConfUtils.maxRetryDurationMillis(getConf)
   lazy private val timeoutInSeconds = ConfUtils.timeoutInSeconds(getConf)
+  lazy private val proxyConfigOpt = ConfUtils.getProxyConfig(getConf)
 
   lazy private val httpClient = {
+
     val maxConnections = ConfUtils.maxConnections(getConf)
     val config = RequestConfig.custom()
       .setConnectTimeout(timeoutInSeconds * 1000)
       .setConnectionRequestTimeout(timeoutInSeconds * 1000)
       .setSocketTimeout(timeoutInSeconds * 1000).build()
-    HttpClientBuilder.create()
+
+    val clientBuilder = HttpClientBuilder.create()
       .setMaxConnTotal(maxConnections)
       .setMaxConnPerRoute(maxConnections)
       .setDefaultRequestConfig(config)
       // Disable the default retry behavior because we have our own retry logic.
       // See `RetryUtils.runWithExponentialBackoff`.
       .disableAutomaticRetries()
-      .build()
+
+    // Set proxy if provided
+    proxyConfigOpt.foreach { proxyConfig =>
+      val proxy = new HttpHost(proxyConfig.host, proxyConfig.port)
+      clientBuilder.setProxy(proxy)
+
+      // Set credentials if provided
+      if (proxyConfig.username.isDefined && proxyConfig.password.isDefined) {
+        val credsProvider = new BasicCredentialsProvider()
+        credsProvider.setCredentials(
+          new AuthScope(proxy),
+          new UsernamePasswordCredentials(proxyConfig.username.get, proxyConfig.password.get)
+        )
+        clientBuilder.setDefaultCredentialsProvider(credsProvider)
+      }
+
+      if (proxyConfig.noProxyHosts.nonEmpty) {
+        val routePlanner = new DefaultRoutePlanner(DefaultSchemePortResolver.INSTANCE) {
+          override def determineRoute(target: HttpHost,
+                                      request: HttpRequest,
+                                      context: HttpContext): HttpRoute = {
+            if (proxyConfig.noProxyHosts.contains(target.getHostName)) {
+              // Direct route (no proxy)
+              new HttpRoute(target)
+            } else {
+              // Route via proxy
+              new HttpRoute(target, null, proxy, true)
+            }
+          }
+        }
+        clientBuilder.setRoutePlanner(routePlanner)
+      }
+    }
+    clientBuilder.build()
   }
```

Comment:
Is there a way to unit test this?

Reply:
I will add unit test, I was waiting to agree on the general approach before adding tests.

Reply:
added unit test with a locally hosted http server